### PR TITLE
Add April 2022 Pet Mod of the Season

### PR DESCRIPTION
**Pull request by scoopapa**
Adding Restrictions and Roulettemons Random Battles

I wanted to ask about the possibility of hosting our sprites, particularly for the Roulettemons Random Battles format. Being that it's a random battle format it would be 100% functional (and probably get a lot more play from outside of our own community) if it had sprites. Please let me know if that would be okay, and whatever you would need from me.

### CODE REVIEW

File: data/mods/roulettemons/random-teams.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/* eslint max-len: ["error", 240] */
+
+import {PRNG, PRNGSeed} from '../../../sim/prng';
```
Comment (AnnikaCodes) [inline data/mods/roulettemons/random-teams.ts:3]:
```suggestion
import {PRNG, PRNGSeed} from '../../../sim/prng';
```
Please don't override eslint errors on a per-file basis. I know TI did this, but it's no longer something we're doing.

File: data/mods/roulettemons/random-teams.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		const roulettemons = [
+			'koatric', 'aquazelle', 'salamalix', 'brawnkey', 'stuneleon', 'chillyte',
+			'eartharoo', 'crazefly', 'electritar', 'aquatopus', 'scorpita', 'baloon',
+			'kinesel', 'glacida', 'pidgeotine', 'gorilax', 'albatrygon', 'chillvark',
+			'komodith', 'giranium', 'flamyle', 'voltecta', 'ostria', 'ninjoth', 'herbigator',
+			'anteros', 'gladiaster', 'hyperoach', 'barracoth', 'toados', 'voltarak',
+			'mosqung', 'flamepion', 'hyenix', 'rhinolite', 'bellena', 'falcola', 'beanium',
+			'lemotic', 'biceon', 'skeleray', 'specyte', 'ramron', 'panthee', 'blastora',
+			'balar', 'dropacle', 'fluffora', 'dolphena', 'tigire', 'catelax', 'chillytemega',
+		];
```
Comment (AnnikaCodes) [inline data/mods/roulettemons/random-teams.ts:27]: Why? You already list all the roulettemons in the formats-data file; why not use `dex.species.all()` and just filter out Pokémon without `randomBattleMoves`?

File: data/mods/roulettemons/random-teams.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+				this.format.gameType !== 'singles', this.dex.formats.getRuleTable(this.format).has('dynamaxclause'));
+
+			// Okay, the set passes, add it to our team
+			pokemon.push(set);
+			if (pokemon.length === this.maxTeamSize) {
+				// Set Zoroark's level to be the same as the last Pokemon
+				const illusion = teamDetails.illusion;
+				if (illusion) pokemon[illusion - 1].level = pokemon[this.maxTeamSize - 1].level;
+
+				// Don't bother tracking details for the last Pokemon
```
Comment (AnnikaCodes) [inline data/mods/roulettemons/random-teams.ts:93]: If no Pokémon in this format have Illusion, you can remove this.

File: data/mods/roulettemons/random-teams.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			if (set.moves.includes('stickyweb')) teamDetails.stickyWeb = 1;
+			if (set.moves.includes('toxicspikes')) teamDetails.toxicSpikes = 1;
+			if (set.moves.includes('defog')) teamDetails.defog = 1;
+			if (set.moves.includes('rapidspin')) teamDetails.rapidSpin = 1;
+			if (set.moves.includes('auroraveil') || (set.moves.includes('reflect') && set.moves.includes('lightscreen'))) {
+				teamDetails.screens = 1;
+			}
+
+			// For setting Zoroark's level
+			if (set.ability === 'Illusion') teamDetails.illusion = pokemon.length;
```
Comment (AnnikaCodes) [inline data/mods/roulettemons/random-teams.ts:130]: See above re: Zoroark.

**Review by AnnikaCodes — COMMENTED** [review]
Is there a Smogon thread for this? I reviewed the random team generation part; have you tested it locally?

@Marty-D is probably the right person to talk to about sprites.

File: config/formats.ts [hunk 5]
```diff
@@ -560,63 +560,41 @@ export const Formats: FormatList = [
 		section: "Pet Mods",
 	},
 	{
-		name: "[Gen 8] Alternatium",
-		desc: `A metagame made up of only Pok&eacute;mon with alternate forms, with all of them being separate and unique.`,
+		name: "[Gen 8] Restrictions",
+		desc: `<b>Restrictions</b>: A metagame made up of brand new Pok&eacute;mon that are made according to various random and non-random restrictions.`,
```
Comment (KrisXV) [inline config/formats.ts:564]:
```suggestion
		desc: `A metagame made up of brand new Pok&eacute;mon that are made according to various random and non-random restrictions.`,
```

File: config/formats.ts [hunk 6]
```diff
@@ -560,63 +560,41 @@ export const Formats: FormatList = [
...
 		threads: [
-			`&bullet; <a href="https://www.smogon.com/forums/threads/3683767/">Alternatium</a>`,
-			`&bullet; <a href="https://docs.google.com/spreadsheets/d/1bvvkPg1CrUBJFJJeuwkts8elfJcEcahGOoHm-vGBXOI/edit?usp=sharing">Spreadsheet</a>`,
+			`&bullet; <a href="https://www.smogon.com/forums/threads/3673824/">Restrictions on Smogon Forums</a>`,
```
Comment (KrisXV) [inline config/formats.ts:566]:
```suggestion
			`&bullet; <a href="https://www.smogon.com/forums/threads/3673824/">Restrictions</a>`,
```

File: config/formats.ts [hunk 7]
```diff
@@ -560,63 +560,41 @@ export const Formats: FormatList = [
...
-		unbanlist: [
-			'Absol', 'Absol-Mega', 'Appletun', 'Articuno', 'Aurorus', 'Basculin', 'Beartic', 'Bisharp', 'Calyrex', 'Camerupt', 'Camerupt-Mega',
-			'Celesteela', 'Cloyster', 'Crawdaunt', 'Cursola', 'Dedenne', 'Delibird', 'Eelektross', 'Espeon', 'Flapple', 'Flareon', 'Flygon',
-			'Galvantula', 'Girafarig', 'Glaceon', 'Goodra', 'Gourgeist', 'Grimmsnarl', 'Guzzlord', 'Heliolisk', 'Hitmonchan', 'Hitmonlee',
-			'Hitmontop', 'Incineroar', 'Indeedee', 'Jolteon', 'Kangaskhan', 'Kommo-o', 'Lanturn', 'Leafeon', 'Leavanny', 'Ludicolo', 'Lunatone',
-			'Luvdisc', 'Machamp', 'Mandibuzz', 'Medicham', 'Nihilego', 'Palossand', 'Pidgeot', 'Pidgeot-Mega', 'Probopass', 'Pyroar', 'Sawk',
-			'Sceptile', 'Sceptile-Mega', 'Scolipede', 'Shiftry', 'Simisage', 'Slowbro', 'Slowbro-Mega', 'Slowking', 'Swoobat', 'Sylveon',
-			'Tapu Bulu', 'Tapu Fini', 'Tauros', 'Typhlosion', 'Umbreon', 'Ursaring', 'Vaporeon', 'Whiscash', 'Wishiwashi', 'Zebstrika', 'Absolite',
-			'Pidgeotite', 'Sceptilite', 'Cameruptite', 'Slowbronite',
+			`&bullet; <a href="https://www.smogon.com/forums/threads/3649106/">Roulettemons on Smogon Forums</a>`,
```
Comment (KrisXV) [inline config/formats.ts:583]:
```suggestion
			`&bullet; <a href="https://www.smogon.com/forums/threads/3649106/">Roulettemons</a>`,
```

**Review by KrisXV — CHANGES_REQUESTED** [review]

File: data/mods/roulettemons/random-teams.ts (hunk 2 above)
Comment (scoopapa) [reply data/mods/roulettemons/random-teams.ts]: Vanilla mons still had random battle moves (did I check the wrong place?) but filtering for 'heightm' gave the right list of mons.

**Comment by scoopapa** [comment]
> Is there a Smogon thread for this? I reviewed the random team generation part; have you tested it locally?

Just the roulettemons thread: 
https://www.smogon.com/forums/threads/roulettemons.3649106/

I did test the random battle team generation so as far as I know it should work.

**Pull request merged**